test/test_xacro.py

```diff
@@ -1119,6 +1119,36 @@ def test_yaml_support(self):
             self.assert_matches(self.quick_xacro(src, cli=['type:=%s' % i]),
                                 res.format(tag=i))
 
+    def test_yaml_exist_required(self):
+        src = '''
+<a xmlns:xacro="http://www.ros.org/wiki/xacro">
+  <xacro:include filename="${load_yaml('non-existent.yaml')}"/>
+</a>'''
+        self.assertRaises(xacro.XacroException, self.quick_xacro, src)
+
+    def test_yaml_exist_optional(self):
+        src = '''
+<a xmlns:xacro="http://www.ros.org/wiki/xacro">
+  <xacro:include filename="${load_yaml('non-existent.yaml')}" optional="True"/>
```

Comment:
Just noticed that this doesn't make sense. `load_yaml()` returns a python dictionary, which is assigned to `filename` here. You need to pass the `optional` argument to `load_yaml`, like this:
`<xacro:property name="dictionary" value="load_yaml('non-existent.yaml', optional=False)"/>`

Reply:
As of now i have just put the optional parameter to the `include` attribute. As you suggest is an argument to the function `load_yaml()`. Actually I started doing it like that because my use case is mostly `.yaml` files but that would create two different interfaces for the xacro files and yaml files.

Reply:
I'm afraid that I don't really understand your use case. Assigning the result of `load_yaml` (which is a dictionary) to the filename attribute of `xacro:include` doesn't make sense at all to me. Can you provide a real-world use case of yours?

Reply:
You are right, that test case does not make sense. I did it because it was a easy easy way of triggering a yaml case.

I have dropped now the yaml support because is going through the `property` tag instead of `include` and we can do that in another PR.